### PR TITLE
Fix compile problem when open -Wnon-virtual-dtor compile flag

### PR DESCRIPTION
**Pull request by csy0225**
c518a2fdd0b262f7c9174b22f04e5640

### CODE REVIEW

**Review by nosovmik — APPROVED** [review]
Looks good to me, thanks a lot for this finding!

**Comment by apankratovantonp** [comment]
We do not use virtual destruction for all of this class using a pointer to the base class. For some of them, we even move nonvirtual destructor of base class into protected section. The check for this places os false positive.

File: src/core/include/openvino/op/util/read_value_base.hpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ class OPENVINO_API ReadValueBase : public Op, public VariableExtension {
     BWDCMP_RTTI_DECLARATION;
 
     ReadValueBase() = default;
+    virtual ~ReadValueBase() {};
```
Comment (nosovmik) [inline src/core/include/openvino/op/util/read_value_base.hpp:19]: Is this change needed? This class is derived from VariableExtension, if base class has virtual destructor - no need to specify this in derived classes. If it is needed, probably current code style will require to use `~ReadValueBase() override {}` style

File: src/core/include/openvino/op/util/assign_base.hpp [hunk 2]
```diff
@@ -15,6 +15,7 @@ class OPENVINO_API AssignBase : public Sink, public VariableExtension {
     OPENVINO_OP("AssignBase", "util", ov::op::Sink);
     BWDCMP_RTTI_DECLARATION;
     AssignBase() = default;
+    virtual ~AssignBase() {};
```
Comment (nosovmik) [inline src/core/include/openvino/op/util/assign_base.hpp:18]: Same comment as to ReadValueBase

File: src/core/include/openvino/op/util/read_value_base.hpp (hunk 1 above)
Comment (csy0225) [reply src/core/include/openvino/op/util/read_value_base.hpp]: Deleted

File: src/core/include/openvino/op/util/assign_base.hpp (hunk 2 above)
Comment (csy0225) [reply src/core/include/openvino/op/util/assign_base.hpp]: Deleted

**Review by apankratovantonp — APPROVED** [review]

File: src/core/include/openvino/core/any.hpp [hunk 3]
```diff
@@ -541,6 +541,8 @@ class OPENVINO_API Any {
         template <typename... Args>
         Impl(Args&&... args) : value(std::forward<Args>(args)...) {}
 
+        virtual ~Impl() {};
```
Comment (nosovmik) [inline src/core/include/openvino/core/any.hpp:544]:
```suggestion
        virtual ~Impl(){};
```

File: src/core/include/openvino/op/util/variable_extension.hpp [hunk 4]
```diff
@@ -38,6 +38,9 @@ class OPENVINO_API VariableExtension {
     /// \brief Returns the identifier of corresponding variable.
     virtual std::string get_variable_id() const = 0;
 
+protected:
+    virtual ~VariableExtension() {};
```
Comment (nosovmik) [inline src/core/include/openvino/op/util/variable_extension.hpp:42]:
```suggestion
    virtual ~VariableExtension(){};
```

**Review by nosovmik — APPROVED** [review]

File: src/core/include/openvino/core/any.hpp (hunk 3 above)
Comment (nosovmik) [reply src/core/include/openvino/core/any.hpp:544]: Please apply this suggestion to fix code style

**Review by ilyachur — APPROVED** [review]
